docs/content/latest/api/ysql/datatypes/type_array/functions-operators/properties.md

```diff
@@ -0,0 +1,417 @@
+
+```postgresql
+select
+  array_length(arr_1, 1) as arr_1_len,
+  array_length(arr_2, 1) as arr_2_len_1,
+  array_length(arr_2, 2) as arr_2_len_2
+from t where k = 1;
+```
+It produces this result:
+```
+ arr_1_len | arr_2_len_1 | arr_2_len_2 
+-----------+-------------+-------------
+         8 |           2 |           4
+```
+### cardinality()
+
+**Purpose:** Returns the total number of values in the specified array. Notice that the value that this function returns can be computed as the product of values returned by the `array_length()` function along each direction.
+
+**Signature:**
+```
+input value:       anyarray
+return value:      int
+```
+**Example:**
+
+```postgresql
+select
+  cardinality(arr_1) as card_1,
+  cardinality(arr_2) as card_2
+from t where k = 1;
+```
+It produces this result:
+```
+ card_1 | card_2 
+--------+--------
+      8 |      8
+```
+### array_dims()
+
+**Purpose:** Returns a text representation of the same information as `array_lower()` and `array_length()` return, for all dimensions, in a single text value.
+
+**Signature:**
+```
+input value:       anyarray
+return value:      text
+```
+**Example:**
+The `array_dims()` function is useful to produce a result that is easily humanly readable. If you want to use the information that it returns programmatically, then you should use `array_lower()`, `array_upper()`, or `array_length()`.
+
+```postgresql
+select
+  array_dims(arr_1) as arr_1_dims,
+  array_dims(arr_2) as arr_2_dims
+from t where k = 1;
+```
+It produces this result:
+```
+ arr_1_dims | arr_2_dims 
+------------+------------
+ [3:10]     | [2:3][4:7]
+```
+
+## Joint semantics
+
+Create the procedure `assert_semantics_and_traverse_values()` and then invoke it for each of the three provided data sets. You supply it with a one-dimensional array and a two dimensional array and, for each, your humanly determined estimates of these values:
+
+- what `array_ndims()` returns,
+- what `array_lower()` returns for each dimension,
+- what `array_upper()` returns for each dimension.
+
+The procedure obtains the actual values, programmatically, for all of the values that you supply and asserts that they agree.
+
+Notice that the cardinality and the length along each dimension are omitted, by design, from this list. The procedure also obtains these values, programmatically, and checks that these agree with the values that are computed from, respectively, the length along each dimension and the upper and lower bounds along each dimension.
+
+Notice, too, that the value that `array_dims()` returns can be computed from the upper and lower bounds along each dimension. So the procedure does this too and checks that the value returned by `array_dims()` is consistent with the values returned by `array_lower()` and `array_upper()`.
+
+The procedure has some particular requirements:
+
+- the cardinality of each of the two supplied arrays must be the same;
```

Comment:
```suggestion
- the cardinality of each of the two supplied arrays must be the same.
```

Reply:
Now suddenly you want a comma. earlier you wanted a period. Are you a semicolon hater?

Reply:
Nobody will even notice except we too, So I'm accepting it—inconsistency ntwithstandinh.